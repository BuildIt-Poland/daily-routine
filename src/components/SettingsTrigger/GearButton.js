import React from 'react';
import PropTypes from 'prop-types';

import Hitbox from '../Hitbox';
import { Gear } from '../Icons';

function GearButton({ onClick = () => {} }) {
  return (
    <Hitbox aria-label="Open settings" onClick={onClick} data-testid="gear-button">
      <Gear />
    </Hitbox>
  );
}

GearButton.propTypes = {
  onClick: PropTypes.func
};

export default GearButton;
