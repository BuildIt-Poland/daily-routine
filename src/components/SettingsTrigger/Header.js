import React from 'react';

import Subheadline from '../Subheadline';

function Header() {
  return (
    <Subheadline>
      My role <strong>settings</strong>
    </Subheadline>
  );
}

export default Header;
