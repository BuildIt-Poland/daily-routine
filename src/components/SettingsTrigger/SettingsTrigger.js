import React, { useState } from 'react';

import GearButton from './GearButton';
import SidePanel from '../SidePanel';
import Subheadline from '../Subheadline';

function SettingsTrigger() {
  const [isVisible, setIsVisible] = useState(false);

  const toogleSettings = () => setIsVisible(!isVisible);

  const Header = () => {
    return (
      <Subheadline>
        My role <strong>settings</strong>
      </Subheadline>
    );
  };

  return (
    <>
      <GearButton onClick={toogleSettings} />
      <SidePanel header={<Header />} isVisible={isVisible} onClose={toogleSettings} />
    </>
  );
}

export default SettingsTrigger;
