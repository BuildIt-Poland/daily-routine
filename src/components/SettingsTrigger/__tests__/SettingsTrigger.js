import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';

import SettingsTrigger from '../SettingsTrigger';

afterEach(cleanup);

describe('COMPONENT - SettingsTrigger', () => {
  it('renders closed SettingsTrigger', () => {
    const { container } = render(<SettingsTrigger />);

    expect(container).toMatchSnapshot();
  });
  it('renders SettingsTrigger component after clicking GearButton', () => {
    const { getByTestId, container } = render(<SettingsTrigger />);

    fireEvent.click(getByTestId('gear-button'));

    expect(container.querySelector('h2')).toHaveTextContent('My role settings');
  });
});
