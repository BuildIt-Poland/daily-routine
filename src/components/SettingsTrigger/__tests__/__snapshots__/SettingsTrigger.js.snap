// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - SettingsTrigger renders closed SettingsTrigger 1`] = `
.c0 {
  border: 0;
  margin: 0;
  padding: 0;
  background: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 4rem;
  min-height: 4rem;
  cursor: pointer;
}

<div>
  <button
    class="c0"
    data-testid="gear-button"
  >
    <svg
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M24 14.035v-4.07l-2.597-.438a9.677 9.677 0 0 0-1.006-2.424l1.528-2.148-2.878-2.878-2.148 1.528a9.603 9.603 0 0 0-2.424-1.006L14.035 0h-4.07l-.439 2.597a9.658 9.658 0 0 0-2.424 1.007L4.954 2.076 2.076 4.954l1.528 2.148a9.603 9.603 0 0 0-1.006 2.424L0 9.965v4.071l2.597.438a9.677 9.677 0 0 0 1.006 2.424l-1.528 2.148 2.878 2.878 2.148-1.528c.75.439 1.563.781 2.424 1.006L9.965 24h4.071l.438-2.597a9.677 9.677 0 0 0 2.424-1.006l2.148 1.528 2.878-2.878-1.528-2.148a9.603 9.603 0 0 0 1.006-2.424l2.598-.44zM16.629 12a4.629 4.629 0 1 1-9.258 0 4.629 4.629 0 0 1 9.258 0z"
        fill="#fff"
      />
    </svg>
  </button>
</div>
`;
